File: practise/three.cpp
#include<iostream>
using namespace std;

class Demo{
public:
	int a;
private:
	int b;
protected:
	int c;

public:
	void setValues(int x, int y, int z){
		a= x;
		b=y;
		c=z;
	}

	void display(){
		cout << "a = " << a << " b = " << b << " c = " << c << endl;
	}
};

int main(){
	Demo obj;
	obj.a =10;
	// obj.b =20;
	// obj.c =30;

	obj.setValues(10,20,30);
	obj.display();
	return 0;
}

File: practise/three.rb
class Demo
	attr_accessor :a    # public

	def initialize
		@a = @b = @c = nil
	end

	def set_values(x, y, z)
		@a = x
		@b = y
		@c = z
	end

	def display
		puts "a = #{@a} b = #{@b} c = #{@c}"
	end

	private
	attr_accessor :b    # b's getter/setter are private

	protected
	attr_accessor :c    # c's getter/setter are protected
end

# Usage
obj = Demo.new
obj.a = 10
# obj.b = 20   # NoMethodError: private method `b=' called
# obj.c = 30   # NoMethodError / protected method `c=' called

obj.set_values(10, 20, 30)
obj.display
